src/crimson/os/seastore/onode_manager/staged-fltree/stages/key_layout.h

```diff
@@ -42,14 +42,13 @@ template<> struct _full_key_type<KeyT::HOBJ> { using type = key_hobj_t; };
 template <KeyT type>
 using full_key_t = typename _full_key_type<type>::type;
 
-static laddr_t get_lba_hint(shard_t shard, pool_t pool, crush_hash_t crush)
-{
-  if (shard == shard_id_t::NO_SHARD) {
-    return (uint64_t)(pool & 0xFF)<<56 | (uint64_t)(crush)<<24;
-  } else {
-    return (uint64_t)(shard & 0X7F)<<56 | (uint64_t)(pool& 0xFF)<<48 |
-	   (uint64_t)(crush)<<16;
-  }
+static laddr_t get_lba_hint(shard_t shard, pool_t pool, crush_hash_t crush) {
+  // FIXME: It is possible that PGs from different pools share the same prefix
+  // if the mask 0xFF is not long enough, result in unexpected transaction
+  // conflicts.
+  return ((uint64_t)(shard & 0XFF)<<56 |
+          (uint64_t)(pool  & 0xFF)<<48 |
+          (uint64_t)(crush       )<<16);
```

Comment:
NO_SHARD is -1, so it is always 0XFF for the 63~56 bits then hint range is limited.

Reply:
The goal here is to make sure the hint of NO_SHARD oid won't be overlapped with other oids with valid shard, if they co-exist.

Reply:
I remember shard used in EC, replicator will not use shard, so they will not co-exist for one osd instance.

Reply:
Sorry I think I was confused by whether this `shard` stands for an OSD shard (in the m:n model) or an ec shard. I'm good to revert the commit https://github.com/ceph/ceph/pull/43254/commits/0c1c972866bb5522093f6693e212c517c98e7369 with proper comments if @athanatos also think so.

Reply:
Well, we will eventually extend crimson and therefore seastore to support EC pools where the shard field won't necessarily be -1.  I think we'll probably end up expanding the lba address width before then, though.  I'm fine with skipping the shard field for now.